client/components/app.jsx

```diff
@@ -1,29 +1,49 @@
 import React from 'react';
 import Header from './header';
 import GradeTable from './gradeTable';
+import GradeForm from './gradeForm';
 
 class App extends React.Component {
   constructor(props) {
     super(props);
     this.state = { grades: [] };
+    this.addNewGrade = this.addNewGrade.bind(this);
+  }
+
+  componentDidMount() {
+    const fetchRequest = fetch('/api/grades');
+    fetchRequest.then(response => {
+      response.json().then(data => {
+        this.setState({ grades: data });
+      });
+    });
   }
 
   getAverageGrade() {
     const gradesArray = this.state.grades;
     let total = 0;
     let avgGrade = 0;
     for (let i = 0; i < gradesArray.length; i++) {
-      total += gradesArray[i].grade;
+      total += parseInt(gradesArray[i].grade);
     }
     avgGrade = Math.ceil(total / gradesArray.length);
     return avgGrade.toString();
   }
 
-  componentDidMount() {
-    const fetchRequest = fetch('/api/grades');
+  addNewGrade(newGrade) {
+    const url = '/api/grades';
+    const init =
+    {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify(newGrade, null, 2)
```

Comment:
You can leave out the `null`, `2` here. It's just for readability purposes when we actually want to see the data. The server will format the `db.json` file automatically.